From dcdfb8ede3580cde6acc1c6ca889ad3b610d75dc Mon Sep 17 00:00:00 2001
From: Janne Grunau <janne-libav@jannau.net>
Date: Thu, 25 Oct 2012 15:10:22 +0200
Subject: [PATCH 027/278] pcmdec: change default of channels parameter to 1

This was previously implicitly done in ff_raw_read_header().
Fixes fate tests with pcm input.
---
 libavformat/pcmdec.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/pcmdec.c b/libavformat/pcmdec.c
index 40446c2..de6f13e 100644
--- a/libavformat/pcmdec.c
+++ b/libavformat/pcmdec.c
@@ -83,7 +83,7 @@ static int raw_read_packet(AVFormatContext *s, AVPacket *pkt)
 
 static const AVOption pcm_options[] = {
     { "sample_rate", "", offsetof(RawAudioDemuxerContext, sample_rate), AV_OPT_TYPE_INT, {.i64 = 0}, 0, INT_MAX, AV_OPT_FLAG_DECODING_PARAM },
-    { "channels",    "", offsetof(RawAudioDemuxerContext, channels),    AV_OPT_TYPE_INT, {.i64 = 0}, 0, INT_MAX, AV_OPT_FLAG_DECODING_PARAM },
+    { "channels",    "", offsetof(RawAudioDemuxerContext, channels),    AV_OPT_TYPE_INT, {.i64 = 1}, 0, INT_MAX, AV_OPT_FLAG_DECODING_PARAM },
     { NULL },
 };
 
-- 
1.7.9.5

